test/python/test_rticonnextdds_data_access.py

```diff
@@ -380,8 +380,9 @@ def test_get_dictionary(self, populated_input):
       sample.get_dictionary("my_double")
     with pytest.raises(rti.Error, match=r".*invalid TypeCode kind DDS_TK_BOOLEAN.*") as excinfo:
       sample.get_dictionary("my_optional_bool")
-    with pytest.raises(rti.Error, match=r".*invalid TypeCode kind DDS_TK_LONG.*") as excinfo:
-      sample.get_dictionary("my_optional_long")
+    # We return None for unset optional members (that check is done before binding
+    # so this case will return None instead of failing as above).
+    assert None is sample.get_dictionary("my_optional_long")
```

Comment:
It can be done yes, in lua_binding.1.0 it would involve doing the bind before checking if a member is an unset optional. I have it the way round it is currently because I thought it would probably be a bit faster to get member info than to try to bind (though I didn't confirm this).
Which would your prefer?